src/common/Finisher.cc

```diff
@@ -3,6 +3,11 @@
 
 #include "Finisher.h"
 
+#include "common/debug.h"
+
+// re-include our assert to clobber the system one; fix dout:
```

Comment:
do you happen to know who is including `cassert` or `assert.h`? so the `assert()` in ceph is hidden? AFAICT, no headers in libstdc++ but `cassert` includes `assert.h`.